app/src/main/java/com/humberto/tasky/agenda/data/task/TaskRepositoryImpl.kt

```diff
@@ -29,14 +36,15 @@ class TaskRepositoryImpl @Inject constructor(
 
     override suspend fun createTask(task: AgendaItem.Task): EmptyResult<DataError> {
         return try {
+            val taskEntity = task.toTaskEntity()
+            taskDao.upsertTask(taskEntity)
             val result = safeCall {
-                taskDao.upsertTask(task.toTaskEntity())
                 agendaApi.createTask(task.toTaskRequest())
             }.onError { error ->
-                if(error == DataError.Network.NO_INTERNET) {
-                    taskDao.insertModifiedTask(
-                        task.toModifiedTaskEntity(
-                            modificationType = ModificationType.Created
+                if (error.isRetryable()) {
+                    taskDao.insertTaskPendingSync(
+                        taskEntity.toTaskPendingSyncEntity(
+                            userId = localUserId!!
                         )
```

Comment:
You don’t want the coroutine to be cancelled before scheduling this sync call. Although unlikely, the `viewModelScope` will be cancelled when the VM is cleared as the user navigates away from their current screen. To protect against this, we use the `applicationScope`, which is tied to the lifecycle of the `Application`.